### ISSUE TITLE
Can't get all available buckets.

### ISSUE THREAD

**Issue by Nmishin** [bloomreach/s4cmd#35]
I try to use s4cmd as a s3cmd, something like this:

# ./s4cmd -p /home/user/.s3cfg ls s3 
 i see my bucket's list
# ./s4cmd -p /home/user/.s3cfg-n ls s3 
[Invalid Argument] Invalid parameter: s3, s3 path expected
# ./s4cmd -p /home/user/.s3cfg-n ls s3://* 
[Errno -2] Name or service not known
[Errno -2] Name or service not known thread(s)]
[Errno -2] Name or service not known thread(s)]

**Comment by chouhanyang** [comment]
Yeah, this is a good observation. The reason is that bucket is actually like "domain" in a URL. So it is not part of "path". The API to get all buckets are different than the one that recursive through all keys. This needs special hacks in order to achieve this.
